### PR TITLE
IHOST-1627 remote entities redis

### PR DESCRIPTION
**Pull request by ferranorriols**
Added remote entities using the parameter config file `remote_monitoring`

#### Description of the changes

Add a detailed description and purpose of your changes.
Link the issue it solves (if there is one).

### Author

- [ ] add a risk label after carefully considering the "blast radius" of your changes
- [ ] describe the _intent_ of your changes in the description. don't just rewrite your code in prose
- [ ] assign at least one reviewer

### Reviewer

- [ ] review code for readability
- [ ] verify that high risk behavior changes are well tested
- [ ] check license for any new external dependency
- [ ] ask questions about anything that isn't clear and obvious
- [ ] approve the PR when you consider it's good to merge

### CODE REVIEW

File: src/redis.go [hunk 1]
```diff
@@ -82,6 +86,15 @@ func main() {
 	fatalIfErr(i.Publish())
 }
 
+func entity(i *integration.Integration, args *argumentList) (*integration.Entity, error) {
+	if args.RemoteMonitoring {
+		n := fmt.Sprintf("%s:%d", args.Hostname, args.Port)
+		return i.Entity(n, entityRemoteType)
```
Comment (ferranorriols) [inline src/redis.go]: Not sure about this namespace. I was between `redis` or `remote`. What do you understand for `namespace` or `entity type` in entities?

Comment (mariomac) [reply src/redis.go]: I think namespace should be `redis`. Right @ikavanaghNR ?

File: redis-config.yml.sample [hunk 2]
```diff
@@ -7,6 +7,7 @@ instances:
       hostname: localhost
       port: 6379
       keys: '{"0":["<KEY_1>"],"1":["<KEY_2>"]}'
+      remote_monitoring: false #new users should leave remote_monitoring = true
```
Comment (mariomac) [inline redis-config.yml.sample]: I would add more clarification here.

File: src/redis.go [hunk 3]
```diff
@@ -9,17 +10,19 @@ import (
...
-	Keys           sdkArgs.JSON `default:"" help:"List of the keys for retrieving their lengths"`
-	KeysLimit      int          `default:"30" help:"Max number of the keys to retrieve their lengths"`
-	Password       string       `help:"Password to use when connecting to the Redis server."`
+	Hostname         string       `default:"localhost" help:"Hostname or IP where Redis server is running."`
+	Port             int          `default:"6379" help:"Port on which Redis server is listening."`
+	UnixSocketPath   string       `default:"" help:"Unix socket path on which Redis server is listening."`
+	Keys             sdkArgs.JSON `default:"" help:"List of the keys for retrieving their lengths"`
+	KeysLimit        int          `default:"30" help:"Max number of the keys to retrieve their lengths"`
+	Password         string       `help:"Password to use when connecting to the Redis server."`
+	RemoteMonitoring bool         `default:"false" help:"Allows to monitor multiple instances."`
```
Comment (mariomac) [inline src/redis.go]: I would add more clarification in the help.

**Pull request merged**